src/ig/IGExporter.ts

```diff
@@ -690,78 +696,22 @@ export class IGExporter {
   }
 
   /**
-   * Adds the package-list.json file to the IG.  If one already exists, it will be used, otherwise
-   * it will be generated based on the package.json.
+   * Adds the package-list.json file to the IG. Generated based on the Configuration history
+   * field.
    *
    * @param igPath {string} - the path where the IG is exported to
    */
-  private addPackageList(igPath: string): void {
-    // If the user provided an index.md file, use that
-    const inputPackageListPath = path.join(this.igDataPath, 'package-list.json');
-    const outputPath = path.join(igPath, 'package-list.json');
-    if (existsSync(inputPackageListPath)) {
-      let mismatch = false;
-      const inputPackageList = readJSONSync(inputPackageListPath);
-      if (inputPackageList['package-id'] !== this.pkg.packageJSON.name) {
-        logger.error(
-          `package-list.json: package-id value (${inputPackageList['package-id']}) does not match name declared in package.json (${this.pkg.packageJSON.name}).  Ignoring custom package-list.json.`,
-          { file: inputPackageListPath }
-        );
-        mismatch = true;
-      }
-      if (inputPackageList.canonical !== this.pkg.packageJSON.canonical) {
-        logger.error(
-          `package-list.json: canonical value (${inputPackageList.canonical}) does not match canonical declared in package.json (${this.pkg.packageJSON.canonical}).  Ignoring custom package-list.json.`,
-          { file: inputPackageListPath }
-        );
-        mismatch = true;
-      }
-      if (!mismatch) {
-        this.copyAsIs(inputPackageListPath, outputPath);
-        logger.info('Copied ig-data/package-list.json.');
-        return;
-      }
+  addPackageList(igPath: string): void {
+    if (this.pkg.config?.history) {
+      const outputPath = path.join(igPath, 'package-list.json');
+      outputJSONSync(outputPath, this.pkg.config.history, { spaces: 2 });
+      logger.info('Generated package-list.json');
+      this.updateOutputLog(outputPath, [this.configPath], 'generated');
+    } else if (/^https?:\/\/hl7.org\//.test(this.pkg.config?.url)) {
+      logger.warn(
+        'HL7 IGs must have a package-list.json. Please define "history" in config.yaml to generate one.'
+      );
```

Comment:
If someone had an existing `package-list.json` file, it will eventually be imported into `this.pkg.config.history`, right? So eventually we will only reach this warning if there is no package-lock information defined anywhere, right?

Reply:
Yes this should happen only if there is no `package-list.json` or information in the `this.pkg.config.history`, and even then only if it is an HL7 IG.

Reply:
Okay! That makes sense, thanks!